src/d3a/d3a_core/redis_connections/redis_communication.py

```diff
@@ -41,20 +43,27 @@
 ZIP_RESULTS_CHANNEL = "d3a-zip-results"
 ZIP_RESULTS_KEY = "d3a-zip-results-key/"
 
+if TYPE_CHECKING:
+    from d3a.models.area import Area
+
 
 class RedisSimulationCommunication:
-    def __init__(self, simulation, simulation_id, live_events):
-        if simulation_id is None:
-            return
+    def __init__(self, simulation, live_events):
         self._live_events = live_events
-        self._simulation_id = simulation_id
+        self._simulation_id = d3a.constants.CONFIGURATION_ID
         self._simulation = simulation
         self._sub_callback_dict = {
-            self._simulation_id + "/stop": self._stop_callback,
-            self._simulation_id + "/pause": self._pause_callback,
-            self._simulation_id + "/resume": self._resume_callback,
-            self._simulation_id + "/live-event": self._live_event_callback,
-            self._simulation_id + "/bulk-live-event": self._bulk_live_event_callback}
+            f"{self._simulation_id}/area-map/": self._area_map_callback
```

Comment:
Why is this allowed to be here even if there is no simulation_id there, whereas other endpoints are shut down?

Reply:
Rest of the endpoints are shutdown for local run when `self._simulation_id` is an empty string and not needed as were designed to only facilitate `stop/pause/resume/live_events` for web-based simulation run , and not for local run. Contrary to that `area-map` is explicitly for local run. Hope its answers your confusion?

Reply:
So where is the code responsible for pausing/stopping simulations run through cli? why isn't it the same for them?

Reply:
On second thought I think all could be subscribed irrespective of local or web based run and irrespective of it being used or not.

Tagging @spyrostz in case he sees any performance degradation in doing so.

Reply:
> So where is the code responsible for pausing/stopping simulations run through cli? 

[Here](https://github.com/gridsingularity/d3a/blob/master/src/d3a/d3a_core/simulation.py#L441)


> why isn't it the same for them?

It's been there even before we joined, as you can see from commit history.  Answering *why* , may be its not handy for user to publish a message to redis for local run `to stop/pause/resume` simulation rather than a simple keypress

Reply:
The CLI code is not really relevant to the Redis channels, because as @faizan2590 mentioned Redis is not used in this flow. In general I agree with the latter observation of Faizan, that these channels are probably better to remain active. They are not used by anyone now, however does not hurt performance-wise to have them, and also might prove useful in the future in case we add such functionalities in the Redis version of the d3a-api-client